### PR TITLE
[INV-101] Deploy frontend

### PR DESCRIPTION
**Pull request by muhammadjon8**
# Pull request for: [101]

## Description
Deploy to the frontend
_Please replace this text with a description of the changes in this PR or attach a screenshots/video of the changes._

## Checklist

- [x] I have manually tested the changes and related features
- [x] I have run the linter and fixed any possible issues
- [x] I have run the tests and fixed any possible issues
- [x] I have run the build and fixed any possible issues

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+FROM node:18 AS build
+
+# Step 2: Set the working directory in the container
+WORKDIR /app
+
+# Step 3: Copy package.json and package-lock.json (or yarn.lock) to install dependencies
+COPY package.json package-lock.json ./
+
+# Step 4: Install dependencies
+RUN npm install
```
Comment (NegatiV-64) [inline Dockerfile]: Use ci instead

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+# Step 5: Copy the rest of the application files
+COPY . .
+
+# Step 6: Build the application for production
```
Comment (NegatiV-64) [inline Dockerfile]: Set NODE ENV as production

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+FROM nginx:alpine
+
+# Step 8: Copy the build files from the previous stage
+COPY --from=build /app/dist /usr/share/nginx/html
+
+# Step 9: Expose port 80 to access the app
+EXPOSE 80
+
+# Step 10: Run the nginx server
+CMD ["nginx", "-g", "daemon off;"]
```
Comment (NegatiV-64) [inline Dockerfile:35]: No need to run Nginx servier

File: nginx.conf
Comment (NegatiV-64) [inline nginx.conf:1]:
```nginx
user nginx;
worker_processes 5;
worker_rlimit_nofile 8192;

error_log  /dev/stderr warn;
pid        /var/run/nginx.pid;

events {
    worker_connections 1024;
}

http {
    # Basic Settings
    sendfile on;
    tcp_nopush on;
    tcp_nodelay on;
    types_hash_max_size 2048;
    server_tokens off;
    ignore_invalid_headers on;

    # Log Formats
    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
                      '$status $body_bytes_sent "$http_referer" '
                      '"$http_user_agent" "$http_x_forwarded_for"';
	  map $status $loggable {
		  ~^[23]  0;
		  default 1;
	  }
    access_log  /dev/stdout  main if=$loggable;

    # MIME Settings
    include /etc/nginx/mime.types;
    default_type application/octet-stream;

    # Limits
    limit_req_zone $binary_remote_addr zone=dos_attack:20m rate=30r/m;

    # Gzip Settings
    gzip on;
    gzip_disable "msie6";
    gzip_vary off;
    gzip_proxied any;
    gzip_comp_level 5;
    gzip_min_length 1000;
    gzip_buffers 16 8k;
    gzip_http_version 1.1;
    gzip_types application/atom+xml application/javascript application/json application/ld+json application/manifest+json application/rss+xml application/vnd.geo+json application/vnd.ms-fontobject application/x-font-ttf application/x-web-app-manifest+json application/xhtml+xml application/xml font/opentype image/bmp image/svg+xml image/x-icon text/cache-manifest text/css text/plain text/vcard text/vnd.rim.location.xloc text/vtt text/x-component text/x-cross-domain-policy;

    server {
        listen 80;
        server_name localhost;
        root /usr/share/nginx/html;
        index index.html;

        # React Router will handle everything
        location / {
            try_files $uri $uri/ /index.html;
        }

        # Only handle 50x errors with Nginx
        error_page   500 502 503 504  /50x.html;
        location = /50x.html {
          root   /usr/share/nginx/html;
        }
    }
}
```

**Review by NegatiV-64 — CHANGES_REQUESTED** [review]

**Comment by muhammadjon8** [comment]
image
I think i completed this part as well

**Review by NegatiV-64 — APPROVED** [review]

**Pull request merged**